context.go

```diff
@@ -0,0 +1,177 @@
+package fronted
+
+import "crypto/x509"
+import "fmt"
+import "net/http"
+import "time"
+import "sync"
+
+import "github.com/getlantern/eventual"
+
+var (
+	contextsMx       sync.Mutex
+	contexts         = make(map[string]*frontingContext)
+	defaultContextID = "default"
+)
+
+// Configure sets the masquerades to use, the trusted root CAs, and the
+// cache file for caching masquerades to set up direct domain fronting
+// in the default context.
+//
+// defaultProviderID is used when a masquerade without a provider is
+// encountered (eg in a cache file)
+func Configure(pool *x509.CertPool, providers map[string]*Provider, defaultProviderID string, cacheFile string) {
+	ConfigureContext(defaultContextID, pool, providers, defaultProviderID, cacheFile)
+}
+
+// ConfigureContext establishes a new independent fronting
+// configuration for each id.  Masquerades are chosen, vetted and
+// used separately from the default configuration and other
+// fronting contexts.
+func ConfigureContext(id string, pool *x509.CertPool, providers map[string]*Provider, defaultProviderID string, cacheFile string) {
```

Comment:
Why use an opaque identifier for contexts rather than making Context a type?

Reply:
hah I started that way, but we tend to pull usage of this out of thin air since it's usually accessed as a global kind of package? it just made it a bit more straightforward in flashlight -- not necessarily what we need to do long term here ...

Reply:
I can move this registry-ish stuff over to flashlight for now if it's preferable.

Reply:
Ah yes, global state as a service :) You can thank me for a lot of that stuff in our various helper packages.

I think it's okay to have a global default context in fronted, but I think for non-default usage it's probably cleaner to make Context a type and let the caller manage the state. Sort of like what `net/http` does with Servers.

Reply:
Sure easy enough. will do.